src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -121,6 +122,37 @@ tnsr::ii<DataVector, 3, Frame> extrinsic_curvature(
 
   return extrinsic_curvature;
 }
+
+template <typename Frame>
+Scalar<DataVector> ricci_scalar(
+    const tnsr::ii<DataVector, 3, Frame>& spatial_ricci_tensor,
+    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector,
+    const tnsr::ii<DataVector, 3, Frame>& extrinsic_curvature,
+    const tnsr::II<DataVector, 3, Frame>& upper_spatial_metric) noexcept {
+  auto ricci_scalar = trace(spatial_ricci_tensor, upper_spatial_metric);
+
+  for (size_t i = 0; i < 3; ++i) {
```

Comment:
1) Precompute K^ij and then you will only need a double loop instead of a quadruple loop

Reply:
This is not obviously faster, because you need a quadruple loop to get K^ij. Larry and I agreed to  leave this as is for now.